src/ArrayObject/LegacyImplementation.php

```diff
@@ -240,9 +247,17 @@ public function getFlags()
      */
     public function getIterator()
     {
-        $class = $this->iteratorClass;
+        $class    = $this->iteratorClass;
+        $iterator = new $class($this->storage);
+
+        if (! $iterator instanceof Iterator) {
+            throw new UnexpectedValueException(sprintf(
+                'Iterator of type %s is not Traversable',
+                $class
+            ));
+        }
```

Comment:
where can storage be set to non iterable? It is typed to array, which is AFAIK always iterable.

Reply:
That would be a BC break, though, as currently we allow providing a class _string_ that is then used to instantiate the internal iterator.

Can definitely consider it for a new major version, but for this one, we  just need to ensure type safety.